tests/wallet/dao_wallet/test_dao_wallets.py

```diff
@@ -0,0 +1,2067 @@
+    for i in range(1, num_blocks):
+        await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(puzzle_hash_0))
+    await full_node_api.wait_for_wallet_synced(wallet_node=wallet_node_0, timeout=30)
+    total_votes = dao_cat_0_bal + dao_cat_1_bal
+
+    assert dao_wallet_0.dao_info.proposals_list[0].amount_voted == total_votes
+    assert dao_wallet_0.dao_info.proposals_list[0].yes_votes == total_votes
+    assert dao_wallet_1.dao_info.proposals_list[0].amount_voted == total_votes
+    assert dao_wallet_1.dao_info.proposals_list[0].yes_votes == total_votes
+
+    # Add a third wallet and check they can find proposal with accurate vote counts and treasury bal
+    dao_wallet_2 = await DAOWallet.create_new_dao_wallet_for_existing_dao(
+        wallet_node_2.wallet_state_manager,
+        wallet_2,
+        treasury_id,
+    )
+    assert dao_wallet_2 is not None
+    assert dao_wallet_2.dao_info.treasury_id == treasury_id
+
+    await full_node_api.wait_for_wallet_synced(wallet_node=wallet_node_0, timeout=30)
+    await full_node_api.wait_for_wallet_synced(wallet_node=wallet_node_1, timeout=30)
+    await full_node_api.wait_for_wallet_synced(wallet_node=wallet_node_2, timeout=30)
+    await time_out_assert(20, len, 1, dao_wallet_2.dao_info.proposals_list)
+    await time_out_assert(20, int, total_votes, dao_wallet_2.dao_info.proposals_list[0].amount_voted)
+    assert (await dao_wallet_2.get_balance_by_asset_type()) == xch_funds
+
+    # Get the proposal from singleton store and check the singleton block height updates correctly
+    proposal_state = await dao_wallet_0.get_proposal_state(prop.proposal_id)
+    assert proposal_state["passed"]
+    assert not proposal_state["closable"]
+    assert proposal_state["blocks_needed"] == 2
+
+    for i in range(1, 5):
+        await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(puzzle_hash_0))
+
+    proposal_state = await dao_wallet_0.get_proposal_state(prop.proposal_id)
+    assert proposal_state["passed"]
+    assert proposal_state["closable"]
+
+    # Create an update proposal
+    new_dao_rules = DAORules(
+        proposal_timelock=uint64(5),
+        soft_close_length=uint64(4),
+        attendance_required=uint64(200000),  # 100%
+        pass_percentage=uint64(10000),  # 100%
+        self_destruct_length=uint64(8),
+        oracle_spend_delay=uint64(2),
+        proposal_minimum_amount=uint64(1),
+    )
+    update_inner = await dao_wallet_0.generate_update_proposal_innerpuz(new_dao_rules)
+    dao_cat_0_bal = await dao_cat_wallet_0.get_votable_balance()
+    proposal_tx = await dao_wallet_0.generate_new_proposal(update_inner, dao_cat_0_bal, uint64(1000))
+    proposal_sb = proposal_tx.spend_bundle
+    await time_out_assert_not_none(5, full_node_api.full_node.mempool_manager.get_spendbundle, proposal_sb.name())
+    await full_node_api.process_spend_bundles(bundles=[proposal_sb])
+
+    # Give the wallet nodes a second
+    await asyncio.sleep(1)
```

Comment:
why a second exactly? isn't there any other way to detecting that nodes have been updated?